src/libosd/cl_dem_uart.c

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <osd/cl_dem_uart.h>
+
+#include <assert.h>
+#include <osd/osd.h>
+#include <stdio.h>
+#include <unistd.h>
+#include "osd-private.h"
+
+static bool is_dem_uart_module(struct osd_hostmod_ctx *hostmod_ctx,
+							   uint16_t dem_uart_di_addr)
```

Comment:
Please configure your editor to not use tabs, and run clang-format on your code to check for the most common style issues.